auth/shared/src/main/scala/io/udash/auth/AuthRequires.scala

```diff
@@ -0,0 +1,15 @@
+package io.udash.auth
+
+trait AuthRequires {
+  /** Checks if user context has required permissions. */
+  def require(permission: PermissionCombinator, requireAuthenticated: Boolean = false)(implicit userCtx: UserCtx): Unit = {
+    if (requireAuthenticated && userCtx.isInstanceOf[UserCtx.Unauthenticated]) throw new UnauthenticatedException()
```

Comment:
This is the 4th implementation of this branching that I've seen so far and this one is actually reusable. Can we only have one?